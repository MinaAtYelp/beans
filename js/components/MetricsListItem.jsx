import React from 'react';


const MetricsListItem = ({ metric }) => {
  if (!metric) {
    return <div>Malformatted</div>;
  }
  return (
    <div className="panel panel-danger">
      <div className="panel-heading">{metric.title}</div>
      <ul className="list-group">
        <li className="list-group-item">
          Total Subscribed: {metric.subscribed}
        </li>
        <li className="list-group-item">
          Total Meetings: {metric.meetings}
        </li>
      </ul>
    </div>
  );
};

MetricsListItem.propTypes = {
  metric: React.PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
};

export default MetricsListItem;
