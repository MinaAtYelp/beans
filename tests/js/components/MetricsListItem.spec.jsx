import React from 'react';
import renderer from 'react-test-renderer';
import MetricsListItem from '../../../js/components/MetricsListItem';


describe('MetricsListItem', () => {
  it('is rendered without data', () => {
    const component = renderer.create(<MetricsListItem />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('is rendered with one subscription', () => {
    const metric = { title: 'weekly', subscribed: 10, meetings: 20 };
    const component = renderer.create(<MetricsListItem metric={metric} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('is rendered with many subscriptions', () => {
    const metric = { title: 'monthly', subscribed: 2, meetings: 4 };
    const component = renderer.create(<MetricsListItem metric={metric} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
