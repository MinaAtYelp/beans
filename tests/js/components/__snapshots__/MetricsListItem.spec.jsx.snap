// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetricsListItem is rendered with many subscriptions 1`] = `
<div
  className="panel panel-danger"
>
  <div
    className="panel-heading"
  >
    monthly
  </div>
  <ul
    className="list-group"
  >
    <li
      className="list-group-item"
    >
      Total Subscribed: 
    </li>
    <li
      className="list-group-item"
    >
      Participants this week: 
    </li>
  </ul>
</div>
`;

exports[`MetricsListItem is rendered with one subscription 1`] = `
<div
  className="panel panel-danger"
>
  <div
    className="panel-heading"
  >
    weekly
  </div>
  <ul
    className="list-group"
  >
    <li
      className="list-group-item"
    >
      Total Subscribed: 
    </li>
    <li
      className="list-group-item"
    >
      Participants this week: 
    </li>
  </ul>
</div>
`;
